src/QsCompiler/BondSchemas/BondSchemaTranslator.cs

```diff
@@ -0,0 +1,1344 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Numerics;
+using Microsoft.Quantum.QsCompiler.DataTypes;
+
+namespace Microsoft.Quantum.QsCompiler.BondSchemas
+{
+    /// <summary>
+    /// This class translates compiler objects to Bond schema objects.
+    /// </summary>
+    internal static class BondSchemaTranslator
+    {
+        /// <summary>
+        /// Creates a Bond schema QsCompilation object from a QsCompilation compiler object.
+        /// </summary>
+        public static QsCompilation CreateBondCompilation(SyntaxTree.QsCompilation qsCompilation) =>
+            new QsCompilation
+            {
+                Namespaces = qsCompilation.Namespaces.Select(n => n.ToBondSchema()).ToList(),
+                EntryPoints = qsCompilation.EntryPoints.Select(e => e.ToBondSchema()).ToList()
+            };
+
+        private static AccessModifier ToBondSchema(this SyntaxTokens.AccessModifier accessModifier) =>
+            accessModifier.Tag switch
+            {
+                SyntaxTokens.AccessModifier.Tags.DefaultAccess => AccessModifier.DefaultAccess,
+                SyntaxTokens.AccessModifier.Tags.Internal => AccessModifier.Internal,
+                _ => throw new ArgumentException($"Unsupported AccessModifier: {accessModifier}")
+            };
+
+        private static CallableInformation ToBondSchema(this SyntaxTree.CallableInformation callableInformation) =>
+            new CallableInformation
+            {
+                Characteristics = callableInformation.Characteristics.ToBondSchema(),
+                InferredInformation = callableInformation.InferredInformation.ToBondSchema()
+            };
+
+        private static Identifier ToBondSchema(this SyntaxTree.Identifier identifier)
+        {
+            string? bondLocalVariable = null;
+            QsQualifiedName? bondGlobalCallable = null;
+            NonNullable<string> compilerLocalVariable = default;
+            SyntaxTree.QsQualifiedName? compilerGlobalCallable = null;
+            IdentifierKind kind;
+            if (identifier.TryGetLocalVariable(ref compilerLocalVariable))
```

Comment:
You could instead express this as 
```suggestion
            if (identifier is SyntaxTree.Identifier.LocalVariable var)
            {
                kind = IdentifierKind.LocalVariable;
                bondLocalVariable = var.Item.Value;
            }            
```

Reply:
Yes, that way we don't need all the `TryGet`* methods. I will change the pattern to remove these helper methods.

Reply:
Thanks!

Reply:
I moved everything to follow this pattern in the latest changes :)